[BUG] CI_PART1 for DBR 14.3 hangs in the nightly pre-release pipeline

**Describe the bug**
The pipeline appears to fail regularly on a race to create the spark event log dir. The pyspark JVM in the xdist worker fails without crashing the parent python process that keeps hanging until being killed by the timeout.

```
INTERNALERROR>         try:
[2025-02-09T07:54:55.460Z] INTERNALERROR> >           assert False, formatted_error
[2025-02-09T07:54:55.460Z] INTERNALERROR> E           AssertionError: Traceback (most recent call last):
[2025-02-09T07:54:55.460Z] INTERNALERROR> E               File "/home/ubuntu/.local/lib/python3.10/site-packages/_pytest/main.py", line 281, in wrap_session
[2025-02-09T07:54:55.460Z] INTERNALERROR> E                 config.hook.pytest_sessionstart(session=session)
[2025-02-09T07:54:55.460Z] INTERNALERROR> E               File "/home/ubuntu/.local/lib/python3.10/site-packages/pluggy/_hooks.py", line 513, in __call__
[2025-02-09T07:54:55.460Z] INTERNALERROR> E                 return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
[2025-02-09T07:54:55.460Z] INTERNALERROR> E               File "/home/ubuntu/.local/lib/python3.10/site-packages/pluggy/_manager.py", line 120, in _hookexec
[2025-02-09T07:54:55.460Z] INTERNALERROR> E                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
[2025-02-09T07:54:55.460Z] INTERNALERROR> E               File "/home/ubuntu/.local/lib/python3.10/site-packages/pluggy/_callers.py", line 139, in _multicall
[2025-02-09T07:54:55.460Z] INTERNALERROR> E                 raise exception.with_traceback(exception.__traceback__)
[2025-02-09T07:54:55.460Z] INTERNALERROR> E               File "/home/ubuntu/.local/lib/python3.10/site-packages/pluggy/_callers.py", line 122, in _multicall
[2025-02-09T07:54:55.460Z] INTERNALERROR> E                 teardown.throw(exception)  # type: ignore[union-attr]
[2025-02-09T07:54:55.460Z] INTERNALERROR> E               File "/home/ubuntu/.local/lib/python3.10/site-packages/_pytest/logging.py", line 782, in pytest_sessionstart
[2025-02-09T07:54:55.460Z] INTERNALERROR> E                 return (yield)
[2025-02-09T07:54:55.460Z] INTERNALERROR> E               File "/home/ubuntu/.local/lib/python3.10/site-packages/pluggy/_callers.py", line 103, in _multicall
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 res = hook_impl.function(*args)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E               File "/home/ubuntu/spark-rapids/integration_tests/src/main/python/spark_init_internal.py", line 146, in pytest_sessionstart
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 .appName('rapids spark plugin integration tests (python)').getOrCreate()
[2025-02-09T07:54:55.461Z] INTERNALERROR> E               File "/databricks/spark/python/pyspark/sql/session.py", line 536, in getOrCreate
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 else SparkContext.getOrCreate(sparkConf)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E               File "/databricks/spark/python/pyspark/context.py", line 563, in getOrCreate
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 SparkContext(conf=conf or SparkConf())
[2025-02-09T07:54:55.461Z] INTERNALERROR> E               File "/databricks/spark/python/pyspark/context.py", line 207, in __init__
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 self._do_init(
[2025-02-09T07:54:55.461Z] INTERNALERROR> E               File "/databricks/spark/python/pyspark/context.py", line 303, in _do_init
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 self._jsc = jsc or self._initialize_context(self._conf._jconf)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E               File "/databricks/spark/python/pyspark/context.py", line 437, in _initialize_context
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 return self._jvm.JavaSparkContext(jconf)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E               File "/databricks/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1620, in __call__
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 return_value = get_return_value(
[2025-02-09T07:54:55.461Z] INTERNALERROR> E               File "/databricks/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-02-09T07:54:55.461Z] INTERNALERROR> E                 raise Py4JJavaError(
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             : java.io.IOException: Target log directory already exists (file:/tmp/spark-events/eventlog_v2_local-1739087636272)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at org.apache.spark.deploy.history.RollingEventLogFilesWriter.start(EventLogFileWriters.scala:319)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at org.apache.spark.scheduler.EventLoggingListener.start(EventLoggingListener.scala:81)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at org.apache.spark.SparkContext.$anonfun$new$43(SparkContext.scala:909)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at com.databricks.spark.util.WarmupUtils$.runOrRegisterAsCallbackOnWarmUpDone(WarmupUtils.scala:251)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at org.apache.spark.SparkContext.<init>(SparkContext.scala:904)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:61)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:397)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at py4j.Gateway.invoke(Gateway.java:257)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:199)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at py4j.ClientServerConnection.run(ClientServerConnection.java:119)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E             	at java.lang.Thread.run(Thread.java:750)
[2025-02-09T07:54:55.461Z] INTERNALERROR> E           assert False
[2025-02-09T07:54:55.461Z] INTERNALERROR> 
[2025-02-09T07:54:55.461Z] INTERNALERROR> ../../../../.local/lib/python3.10/site-packages/xdist/dsession.py:232: AssertionError
```
**Steps/Code to reproduce bug**
This issue is not 14.3 specific and not specific to nightly

It can be reproduced by running something like
```
TESTS=array_test.py ./jenkins/databricks/test.sh |& tee test.log
```

It is probably introduced by #4217 that explicitly sets https://github.com/NVIDIA/spark-rapids/pull/4217/files#diff-db28879431d57d0e454a2c7ee89fdda9abdec463c61771333d6a6565bf96c062R75

which disables automatic location deconflicting using xdist worker id 
https://github.com/NVIDIA/spark-rapids/blob/6e3f6eb2d1cc7ef33348858d0263c7c371a43745/integration_tests/src/main/python/spark_init_internal.py#L215-L217

Even though the dir name includes app id `local-1739087636272` there an increased chance of getting the same id across workers because of the increased degree of parallelism (8 xdist workers) on larger DBR instances in the pipeline.    

**Expected behavior**
Should not hang 

**Additional context**
N/A

--- a/jenkins/databricks/test.sh
+++ b/jenkins/databricks/test.sh
@@ -74,10 +74,6 @@ IS_SPARK_321_OR_LATER=0
 # --packages in distributed setups, should be fixed by
 # https://github.com/NVIDIA/spark-rapids/pull/5646
 
-# Enable event log for qualification & profiling tools testing
-export PYSP_TEST_spark_eventLog_enabled=true
-mkdir -p /tmp/spark-events
-
 export PYSP_TEST_spark_rapids_shims_spark350db143_enabled=${PYSP_TEST_spark_rapids_shims_spark350db143_enabled:-true}
 
 rapids_shuffle_smoke_test() {

